x-pack/metricbeat/module/openai/usage/usage_integration_test.go

```diff
@@ -0,0 +1,177 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build integration
+
+package usage
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	mbtest "github.com/elastic/beats/v7/metricbeat/mb/testing"
+)
+
+func TestFetch(t *testing.T) {
+	apiKey := time.Now().String() // to generate a unique API key everytime
+	usagePath := "/usage"
+	server := initServer(usagePath, apiKey)
+	defer server.Close()
+
+	f := mbtest.NewReportingMetricSetV2Error(t, getConfig(server.URL+"/usage", apiKey))
+
+	events, errs := mbtest.ReportingFetchV2Error(f)
+	if len(errs) > 0 {
+		t.Fatalf("Expected 0 error, has %d: %v", len(errs), errs)
+	}
+
+	assert.NotEmpty(t, events)
+
+}
+
+func TestData(t *testing.T) {
+	apiKey := time.Now().String() // to generate a unique API key everytime
+	usagePath := "/usage"
+	server := initServer(usagePath, apiKey)
+	defer server.Close()
+
+	f := mbtest.NewReportingMetricSetV2Error(t, getConfig(server.URL+"/usage", apiKey))
+
+	err := mbtest.WriteEventsReporterV2Error(f, t, "")
+	if !assert.NoError(t, err) {
+		t.FailNow()
+	}
```

Comment:
[Suggestion]
1 - thre is just one assertion, so why would you need a `FailNow`?
2 - use `require` instead of `assert` when you need to "fail now"
3 - there is a `require|assert.Error`, better than `!NoError`

Reply:
Done!